scijava-types/src/main/java/org/scijava/types/SubTypeExtractor.java

```diff
@@ -41,37 +41,31 @@
  */
 public abstract class SubTypeExtractor<T> implements TypeExtractor {
 
-	protected abstract Class<?> getRawType();
-
 	/**
 	 * Returns a {@link Type} array, where the {@code i}th {@link Type} in the
 	 * array is the {@code i}th type parameter of the {@link Class} returned by
-	 * {@link #getRawType()}. Similar to
+	 * {@link #baseClass()}. Similar to
 	 * {@link ParameterizedType#getActualTypeArguments()}, but works on
 	 * {@link Object}s instead of {@link ParameterizedType}s.
 	 *
 	 * @param r the {@link TypeReifier} used to reify the type parameters.
 	 * @param object the {@link Object} to extract type parameters from.
 	 * @see ParameterizedType#getActualTypeArguments()
 	 * @return the actual type arguments of {@code Object}, with respect to the
-	 *         {@link Type} returned by {@link #getRawType()}
+	 *         {@link Type} returned by {@link #baseClass()}
 	 */
 	protected abstract Type[] getTypeParameters(final TypeReifier r,
 		final T object);
 
-	@Override
-	public boolean canReify(TypeReifier r, Class<?> cls) {
-		return getRawType().isAssignableFrom(cls);
-	}
-
 	@Override
 	public Type reify(final TypeReifier r, final Object object) {
-		if (!canReify(r, object.getClass())) throw new IllegalArgumentException(
-			this + " can only reify Objects of Class " + getRawType()
-				.getSimpleName() + "!");
+		if (!baseClass().isAssignableFrom(object.getClass()))
+			throw new IllegalArgumentException(this +
```

Comment:
Instead of `this +`, should this error message be `this.getClass().getName() +`? I know it's just left alone from how it was before, but do these type extractor plugins actually stringify in the desired way for this error message to read correctly? I'm guessing not...

Reply:
Changed in 2e55efac